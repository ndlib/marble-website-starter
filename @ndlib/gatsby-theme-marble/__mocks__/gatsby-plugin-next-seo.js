const React = require('react')
const plugin = jest.requireActual('gatsby-plugin-next-seo')

module.exports = {
  ...plugin,
  JsonLd: jest.fn().mockImplementation(
    () => null
  )
}
