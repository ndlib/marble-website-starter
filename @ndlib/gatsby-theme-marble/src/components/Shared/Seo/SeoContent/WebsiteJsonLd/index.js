import React from 'react'
import PropTypes from 'prop-types'
import { JsonLd } from '../JsonLd'

export default function WebsiteJsonLd({ pathname, siteUrl }) {
  if (pathname === '/') {
    return (
      <JsonLd>
        {{
          "@context": "https://schema.org",
          "@type": "Website",
          url: {siteUrl},
          potentialAction: {
            "@type": "SearchAction",
            target: `${siteUrl}/search?q={search_term_string}`,
            "query-input": "required name=search_term_string",
          },
        }}
      </JsonLd>
    )
  }
  return null
}

WebsiteJsonLd.propTypes = {
  pathname: PropTypes.string.isRequired,
  siteUrl: PropTypes.string.isRequired,
}
