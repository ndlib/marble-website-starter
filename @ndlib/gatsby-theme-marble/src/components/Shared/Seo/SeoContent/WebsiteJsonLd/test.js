import React from 'react'
import { shallow } from 'enzyme'
import WebsiteJsonLd from './'
import { JsonLd } from 'gatsby-plugin-next-seo'

test('Website JSON LD schema on homepage', () => {
  const wrapper = shallow(<WebsiteJsonLd pathname='/' siteUrl='https://marble.nd.edu' />)
  console.log(wrapper.debug())
  expect(wrapper.find(JsonLd).props().url).toEqual('https://marble.nd.edu')
})
