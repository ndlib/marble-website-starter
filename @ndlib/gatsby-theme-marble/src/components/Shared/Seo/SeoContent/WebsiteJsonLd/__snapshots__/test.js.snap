// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WebsiteJsonLd component renders props into values 1`] = `null`;
